test/groovy/templates/PiperPipelineStageInitTest.groovy

```diff
@@ -139,6 +142,34 @@ class PiperPipelineStageInitTest extends BasePiperTest {
         assertThat(nullScript.commonPipelineEnvironment.configuration.stageStashes.Init.unstash, is([]))
     }
 
+    @Test
+    void testTransportRequestReqIDFromGitIfFalse() {
+        jsr.step.piperPipelineStageInit(
+            script: nullScript,
+            juStabUtils: utils,
+            buildTool: 'maven',
+            stashSettings: 'com.sap.piper/pipeline/stashSettings.yml',
+            transportRequestReqIDFromGit: false
+        )
+        assertThat(stepsCalled, hasItems('checkout', 'setupCommonPipelineEnvironment', 'piperInitRunStageConfiguration', 'artifactPrepareVersion', 'pipelineStashFilesBeforeBuild'))
+        assertThat(stepsCalled, not(hasItems('slackSendNotification')))
+        assertThat(nullScript.commonPipelineEnvironment.configuration.stageStashes.Init.unstash, is([]))
+    }
+
+    @Test
+    void testTransportRequestReqIDFromGitIfTrue() {
+        jsr.step.piperPipelineStageInit(
+            script: nullScript,
+            juStabUtils: utils,
+            buildTool: 'maven',
+            stashSettings: 'com.sap.piper/pipeline/stashSettings.yml',
+            transportRequestReqIDFromGit: true
+        )
+        assertThat(stepsCalled, hasItems('checkout', 'transportRequestReqIDFromGit', 'setupCommonPipelineEnvironment', 'piperInitRunStageConfiguration', 'artifactPrepareVersion', 'pipelineStashFilesBeforeBuild'))
+        assertThat(stepsCalled, not(hasItems('slackSendNotification')))
+        assertThat(nullScript.commonPipelineEnvironment.configuration.stageStashes.Init.unstash, is([]))
```

Comment:
one assert should be fine here:

assertThat(stepsCalled, hasItems('transportRequestReqIDFromGit'))